File: c/7.c
/* http://projecteuler.net/problem=7 */
#include <stdio.h>
#include <math.h>

int is_prime(long num) {
        long i=0;
        int lim=0;

        if (num == 1) {
                return 0;
        }

        if (num == 2) {
                return 1;
        }

        lim = (int)sqrt(num);

        for (i=2 ; i<=lim ; i++) {
                if (num%i == 0)
                        return 0;
        }

        return 1;
}

int main(int argc, char* argv[])
{
        int cnt=0;
        long i=1;

        for (;;) {
                if (is_prime(i))
                        cnt++;
                
                if (cnt == 10001)
                        break;

                i++;
        }

        printf("10001 th prime = %d\n", i);

        return 0;
}

File: python/7.py
# By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can 
# see that the 6th prime is 13.
#
# What is the 10001st prime number?
#
# https://projecteuler.net/problem=7

import string, sys
import commonutils

def main():
    count = 0
    num = 1

    while True:
        if commonutils.isPrime(num):
            count += 1
       
        if count == 10001:
            break

        num += 1

    print("The 10001th prime number = ", num)

if __name__ == '__main__':
    main()
